sites/all/modules/custom/training2_form/training2_form.module

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/**
+ * @file
+ * Homework on topic Block and Menu.
+ */
+
+/**
+ * Secret key to login without username and password.
+ */
+define('TRAINING2_FORM_SECRET_KEY', 'сим-сим откройся');
+
+/**
+ * Url with current exchange rates.
+ */
+define('TRAINING2_FORM_NBRB_RATES_XML', 'http://www.nbrb.by/Services/XmlExRates.aspx');
+
+/**
+ * Implements hook_block_info().
+ */
+function training2_form_block_info() {
+  $blocks['currency'] = array(
+    'info' => t('Currency block'),
+    'cache' => DRUPAL_CACHE_GLOBAL,
+  );
+
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_configure().
+ */
+function training2_form_block_configure($delta = '') {
+  $form = array();
+
+  if ($delta = 'currency') {
+    $currency = training2_form_conf_content();
+
+    $form['form_currency'] = array(
+      '#type' => 'checkboxes',
+      '#title' => t('Choose currencies'),
+      '#options' => drupal_map_assoc($currency),
+    );
+  }
+
+  return $form;
+}
+
+/**
+ * Implements hook_block_save().
+ */
+function training2_form_block_save($delta = '', $edit = array()) {
+
+  if ($delta == 'currency') {
+    if (!empty($edit['form_currency'])) {
+      variable_set('form_currency', $edit['form_currency']);
+    }
+  }
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function training2_form_block_view($delta = '') {
+  $block = array();
+
+  switch ($delta) {
+    case 'currency':
+      $block['subject'] = t('Currency');
+      $block['content'] = training2_form_content();
+      break;
+  }
+
+  return $block;
+}
+
+/**
+ * Function drupal_map_assoc() uses return array.
+
+ * @return array
+ *   Return array which includes CharCode and Rate.
+ */
+function training2_form_conf_content() {
+  $xml_array = training2_form_get_xml_data();
+  $array_currency = $xml_array['Currency'];
+
+  $arr_value = array();
+  $n = count($array_currency);
+  for ($i = 0; $i < $n; $i++) {
+    $arr_value[$i] = $array_currency[$i]['CharCode'] . ' - ' . $array_currency[$i]['Rate'];
+  }
+  return $arr_value;
+}
+
+/**
+ * Transformation XML.
+ *
+ * @return mixed
+ *   Return array currency.
+ */
+function training2_form_get_xml_data() {
+  $xml_array = array();
+  $url = TRAINING2_FORM_NBRB_RATES_XML;
+
+  if (!(empty($url))) {
```

Comment:
Unnecessary condition
Could be rewritten if (defined(TRAINING2_FORM_NBRB_RATES_XML)). However, you can exclude the condition